src/rgw/rgw_rest_pubsub.cc

```diff
@@ -125,9 +181,32 @@ class RGWPSCreateTopicOp : public RGWOp {
   }
 
   public:
-  int verify_permission(optional_yield) override {
-    return 0;
-  }
+   int verify_permission(optional_yield y) override {
+    auto ret = get_params();
+    if (ret < 0) {
+      return ret;
+    }
+
+    const RGWPubSub ps(driver, s->owner.get_id().tenant);
+    rgw_pubsub_topic result;
+    ret = ps.get_topic(this, topic_name, result, y);
+    if (ret == -ENOENT) {
+      // topic not present
+      return 0;
+    }
+    if (ret == 0) {
+      if (result.user == s->owner.get_id() ||
+          !s->cct->_conf->mandatory_topic_permissions) {
+        return 0;
+      }
+      ldpp_dout(this, -1) << "failed to create topic '" << topic_name
```

Comment:
please avoid "-1" debug level. these cannot be turned off

Reply:
> please avoid "-1" debug level. these cannot be turned off

My understanding was error's are "-1" and should not be allowed to supress so -1 ?

Reply:
this is a wider discussion, but, IMO, we always need a way to turn off debug logs. otherwise these errors can cause denial of service.

Reply:
high rate of error logs can slow down the system. it should be possible to turn them off if the user decide that the harm caused by this slowdown is greater than the value these logs provide

Reply:
> > this is a wider discussion, but, IMO, we always need a way to turn off debug logs. otherwise these errors can cause denial of service.
> 
> denial of service ?? not sure if i understand this ? not able to turn of debug/error messages can lead to denial of service ? I mean any S3 operation that is failing, should have an option to turn of that error log IMO is little weird ?

@yuvalif as discussed during the rgw weekly, it was recommend to avoid -1 specially during the http_request that are send, so will remove change the log level to 1 from -1.
thanks